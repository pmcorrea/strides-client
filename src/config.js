export default {
	API_ENDPOINT: "https://young-castle-93107.herokuapp.com/graphql",
	// API_ENDPOINT: "http://localhost:5000/graphql",
	TOKEN_KEY: "Strides-token-key",
};