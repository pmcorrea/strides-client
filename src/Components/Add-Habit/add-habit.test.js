import React from 'react';
import ReactDOM from 'react-dom'
import { BrowserRouter } from 'react-router-dom'
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect'
import AddHabit from './add-habit';

import { ApolloProvider, ApolloClient, createHttpLink, InMemoryCache } from '@apollo/client';
import { setContext } from "apollo-link-context";
import TokenHelpers from "../../Services/token-helpers"
import config from '../../config'

// Apollo Link will send token with every request
const authLink = setContext((_, { headers }) => {
	const token = TokenHelpers.getAuthToken()
	return {
		headers: {
			...headers,
			authorization: token ? `bearer ${token}` : null,
		}
	}
})

const httpLink = new createHttpLink({ uri: `${config.API_ENDPOINT}` })

// Instantiate Apollo Client
const apolloClient = new ApolloClient({
	cache: new InMemoryCache(),
	link: authLink.concat(httpLink)
})

afterEach(cleanup)

// Snapshot Test
it('should take a snapshot', () => {
	const { asFragment } = render(
		<ApolloProvider client={apolloClient}>
	<BrowserRouter>
			<AddHabit />
	</BrowserRouter>
	 </ApolloProvider >
	)

	expect(asFragment(
		<ApolloProvider client={apolloClient}>
		<BrowserRouter>
			<AddHabit />
		</BrowserRouter>
		</ApolloProvider>)
		).toMatchSnapshot()
})

// Smoke Test
it('renders without crashing', () => {
	const div = document.createElement('div')

	ReactDOM.render(
		<ApolloProvider client={apolloClient}>
		<BrowserRouter>
			<AddHabit />
		</BrowserRouter>
		</ApolloProvider>,
		div
	)
	
	ReactDOM.unmountComponentAtNode(div)
})