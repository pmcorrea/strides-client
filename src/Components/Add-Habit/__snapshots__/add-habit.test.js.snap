// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class=""
  >
    <form
      action=""
      class="add-habit-form"
    >
      <p>
        I want to
      </p>
      <input
        name="habit"
        placeholder="meditate"
        type="text"
      />
      <p>
        every
      </p>
      <div>
        <button
          class="circle-button-s"
          style="color: black;"
          type="button"
          value="sunday"
        >
          S
        </button>
        <button
          class="circle-button-s"
          style="color: black;"
          type="button"
          value="monday"
        >
          M
        </button>
        <button
          class="circle-button-s"
          style="color: black;"
          type="button"
          value="tuesday"
        >
          T
        </button>
        <button
          class="circle-button-s"
          style="color: black;"
          type="button"
          value="wednesday"
        >
          W
        </button>
        <button
          class="circle-button-s"
          style="color: black;"
          type="button"
          value="thursday"
        >
          T
        </button>
        <button
          class="circle-button-s"
          style="color: black;"
          type="button"
          value="friday"
        >
          F
        </button>
        <button
          class="circle-button-s"
          style="color: black;"
          type="button"
          value="saturday"
        >
          S
        </button>
      </div>
      <p>
        for 30 days
      </p>
      <button
        type="submit"
      >
        Submit
      </button>
    </form>
    <div
      class="bottom-toolbar"
    >
      <button>
        <a
          href="/home"
        >
          <img
            alt=""
            class="icon-bar-icon"
            src="home-icon.svg"
          />
        </a>
      </button>
      <button>
        <a
          href="/add-habit"
        >
          <img
            alt=""
            class="icon-bar-icon"
            src="plus-icon.svg"
          />
        </a>
      </button>
      <button>
        <a
          href="/profile"
        >
          <img
            alt=""
            class="icon-bar-icon"
            src="profile-icon.svg"
          />
        </a>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
